src/hooks/Orderbook/useDrawOrderbook.ts

```diff
@@ -0,0 +1,369 @@
+    if (price) {
+      ctx.fillStyle = textColor;
+      ctx.fillText(
+        MustBigNumber(price).toFixed(tickSizeDecimals ?? SMALL_USD_DECIMALS),
+        getXByColumn({ canvasWidth, colIdx: 1 }) - ORDERBOOK_ROW_PADDING_RIGHT,
+        y
+      );
+    }
+
+    // Mine text
+    if (mine) {
+      ctx.fillStyle = textColor;
+      ctx.fillText(
+        MustBigNumber(mine).toFixed(stepSizeDecimals ?? TOKEN_DECIMALS),
+        getXByColumn({ canvasWidth, colIdx: 2 }) - ORDERBOOK_ROW_PADDING_RIGHT,
+        y
+      );
+    }
+  };
+
+  const drawOrderbookRow = ({
+    ctx,
+    idx,
+    rowToRender,
+    animationType = OrderbookRowAnimationType.NONE,
+  }: {
+    ctx: CanvasRenderingContext2D;
+    idx: number;
+    rowToRender: PerpetualMarketOrderbookLevel | undefined;
+    animationType?: OrderbookRowAnimationType;
+  }) => {
+    if (!rowToRender) return;
+    const { depth, mine, price, size } = rowToRender;
+    const histogramAccentColor = side === 'bid' ? color.fadedGreen : color.fadedRed;
+
+    // Depth Bar
+    if (depth) {
+      drawBars({
+        barType: 'depth',
+        ctx,
+        depthOrSizeValue: depth,
+        gradientMultiplier: 1.3,
+        histogramAccentColor,
+        histogramSide,
+        idx,
+      });
+    }
+
+    // Size Bar
+    drawBars({
+      barType: 'size',
+      ctx,
+      depthOrSizeValue: size,
+      gradientMultiplier: 5,
+      histogramAccentColor,
+      histogramSide,
+      idx,
+    });
+
+    // Size, Price, Mine
+    drawText({
+      animationType,
+      ctx,
+      idx,
+      mine,
+      price,
+      size,
+    });
+  };
+
+  // Update histograms and row contents on data change
+  useEffect(() => {
+    const ctx = canvas?.getContext('2d');
+
+    if (!canvas || !ctx) return;
+
+    // Clear canvas before redraw
+    ctx.clearRect(0, 0, ctx.canvas.width, ctx.canvas.height);
+
+    // Animate row removal and update
+    const mapOfOrderbookPriceLevels =
+      side && currentOrderbookMap?.[side === 'ask' ? 'asks' : 'bids'];
+    const empty: number[] = [];
+    const removed: number[] = [];
+    const updated: number[] = [];
+
+    prevData.current.forEach((row, idx) => {
+      if (!row) {
+        empty.push(idx);
+        return;
+      }
+
+      if (mapOfOrderbookPriceLevels?.[row.price] === 0) {
+        removed.push(idx);
+        drawOrderbookRow({
+          ctx,
+          idx,
+          rowToRender: row,
+          animationType: OrderbookRowAnimationType.REMOVE,
+        });
+      } else if (mapOfOrderbookPriceLevels?.[row.price] === row?.size) {
+        drawOrderbookRow({
+          ctx,
+          idx,
+          rowToRender: data[idx],
+          animationType: OrderbookRowAnimationType.NONE,
+        });
+      } else {
+        updated.push(idx);
+        drawOrderbookRow({
+          ctx,
+          idx,
+          rowToRender: row,
+          animationType: OrderbookRowAnimationType.NEW,
+        });
+      }
+    });
+
+    setTimeout(() => {
```

Comment:
Thinking about this setTimeout 🤔 
What is the reason we have it? since a new setTimeout is still created on every data update. What happens if we don't have the timeout?